specifyweb/frontend/js_src/lib/utils/fieldFormat.ts

```diff
@@ -80,18 +82,22 @@ function formatValue(
       parseResults,
     });
 
-  return value.toString();
+  return (strict ? undefined : value.toString()) as ReturnType<
+    typeof formatValue<STRICT>
+  >;
 }
 
 /**
  * Like fieldFormat, but synchronous, because it doesn't fetch a pick list if
  * it is not already fetched
  */
-export function syncFieldFormat(
+export function syncFieldFormat<STRICT extends boolean = false>(
   field: LiteralField | undefined,
   parser: Parser | undefined,
-  value: boolean | number | string | null | undefined
-): string {
+  value: boolean | number | string | null | undefined,
+  // @ts-expect-error
+  strict: STRICT = false
```

Comment:
can you clarify why is this needed?

and justify why it is false in most places but has to be true in attachments bulk upload?
i.e can we set it so some value and have that value in all places?

Reply:
This was because this function parses the value, applies all the parsers and validators, and if they fail, it still returns the same  value back. In my case, I want to have it return undefined if the parser fails (due to incorrectly formatted)

Reply:
your solution is valid. I am just asking whether there is a way to simplify, or whether the solution is needed at all.

> and if they fail, it still returns the same value back. In my case, I want to have it return undefined if the parser fails 

1. can you clarify why you need an incorrectly formatted value to return undefined?
2. can you check if we can make it always return undefined - if that change would make sense?

Reply:
1. I'm using that to check if a value is correctly formatted or not. I'm showing an error text ("Incorrectly Formatted") and skipping those files when uploading.
2. Will check the second